artemis-commons/src/main/java/org/apache/activemq/artemis/utils/CompositeAddress.java

```diff
@@ -41,12 +41,35 @@ public static SimpleString toFullyQualified(SimpleString address, SimpleString q
       return result;
    }
 
+   /**
+    * Checks whether a given string complies the fully qualified queue name (FQQN) standard or not.
+    *
+    * @param address The string to check for FQQN standard compliance.
+    * @return boolean
+    */
    public static boolean isFullyQualified(String address) {
-      return address == null ? false : address.contains(SEPARATOR);
+      if (address == null) {
+         return false;
+      }
+
+      int index = address.indexOf(SEPARATOR);
```

Comment:
Does it need to start at 0 or can we safely assume we can use fromIndex 1 given we expect at least on character before. Like wise is it worth checking the length is >= 4 on the expectation there has to be a character before and after the seperator? To return quicker in such cases theres no need to iterate to fail

Reply:
Note it maybe still faster not to have the conditional branches. Im mearly asking if it has been looked into and checked it theres any perf wins like that.  

@franz1981 you maybe able to quickly say if its worth doing or not

Reply:
I can see adding the the validation of at least 4 characters bringing a noticeable performance enhancement and I'll gladly implement it! ;)

Reply:
I just added both these changes, hope everything checks out ;)